presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -256,16 +243,14 @@ private int addNewGroup(long hashPosition, long value)
         groupIds.set(hashPosition, groupId);
 
         // increase capacity, if necessary
-        if (nextGroupId >= maxFill) {
-            rehash();
+        if (needRehash()) {
+            tryRehash();
```

Comment:
+ `verify()` ?

Reply:
we don't know if we can succeed or not right?